megamek/src/megamek/utilities/FileSignerTool.java

```diff
@@ -234,60 +234,67 @@ private PrivateKey loadPrivateKey(File keyFile) throws Exception {
+        return verifyFile(readFileContent(file), readPublicKey(publicKeyFile));
     }
 
     /**
      * Verify a file's signature.
      */
     public static boolean verifyFile(File file, PublicKey publicKey) throws Exception {
-        // Read the file content
-        String content = Files.readString(file.toPath());
-        return verifyFile(content, publicKey);
+        return verifyFile(readFileContent(file), publicKey);
     }
 
     /**
      * Verify a file's signature.
      */
     public static boolean verifyFile(String content, PublicKey publicKey) throws Exception {
-        // Find signature
-        int signatureStartPos = content.lastIndexOf(SIGNATURE_START);
-        if (signatureStartPos == -1) {
-            // logger.debug("No signature found in the content");
-            return false;
-        }
+    // Find signature
+    int signatureStartPos = content.lastIndexOf(SIGNATURE_START);
+    if (signatureStartPos == -1) {
+        // logger.debug("No signature found in the content");
+        return false;
+    }
 
-        int signatureEndPos = content.lastIndexOf(SIGNATURE_END);
-        if (signatureEndPos == -1 || signatureEndPos < signatureStartPos) {
-            // logger.debug("Invalid signature format");
-            return false;
-        }
+    int signatureEndPos = content.lastIndexOf(SIGNATURE_END);
+    if (signatureEndPos == -1 || signatureEndPos < signatureStartPos) {
+        // logger.debug("Invalid signature format");
+        return false;
+    }
 
-        // Extract signature
-        String encodedSignature = content.substring(
-            signatureStartPos + SIGNATURE_START.length(),
-            signatureEndPos
-        );
+    // Extract signature
+    String encodedSignature = content.substring(
+        signatureStartPos + SIGNATURE_START.length(),
+        signatureEndPos
+    );
 
-        byte[] signatureBytes = Base64.getDecoder().decode(encodedSignature);
+    byte[] signatureBytes = Base64.getDecoder().decode(encodedSignature);
 
-        // Get content without signature
-        String contentWithoutSignature = content.substring(0, signatureStartPos);
+    // Get content without signature
+    String contentWithoutSignature = content.substring(0, signatureStartPos);
 
-        // Verify the signature
-        Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
-        signature.initVerify(publicKey);
-        signature.update(contentWithoutSignature.getBytes(StandardCharsets.UTF_8));
+    // Verify the signature
+    Signature signature = Signature.getInstance(SIGNATURE_ALGORITHM);
+    signature.initVerify(publicKey);
+    signature.update(contentWithoutSignature.getBytes(StandardCharsets.UTF_8));
+
+    return signature.verify(signatureBytes);
+    }
+
+    /**
+     * Read file content as a String.
+     */
+    private static String readFileContent(File file) throws Exception {
+        return Files.readString(file.toPath());
```

Comment:
wrapping a function call around a function call to substitute a single function call does not makes sense.